### PR TITLE
feat: replace pr title validator

### PR DESCRIPTION
**Pull request by Xunzhuo**
Signed-off-by: Xunzhuo <mixdeers@gmail.com>

resolve: https://github.com/mosn/layotto/issues/485

**What this PR does**:

replace pr title validator

compared with thehanimo/pr-title-checker:

`amannn/action-semantic-pull-request` is an ideal workflow to check PR Validation, it is followed by [semantic style](
https://github.com/commitizen/conventional-commit-types), as for `thehanimo/pr-title-checker`, we need to write the pattern in json config, and actually thery are not followed by semantic style tyle, and we have to maintain another json config, which is no-needed.

 Also this project is more active than hehanimo/pr-title-checker, which means this project is more stable to be maintained. We should choose a stable and active action as our github workflow, otherwise if any bugs happened, we will get into troubles and hard for us to fix, which leads us to change our actions repeatedly, this is not what maintainers want to see.

### CODE REVIEW

**Comment by Xunzhuo** [comment]
we need give new action: `amannn/action-semantic-pull-request` privilege  to access resource:
image

**Comment by seeflood** [comment]
@ZLBer Any idea on this PR?

**Comment by seeflood** [comment]
@Xunzhuo Hi, why close it?
I can help authorize the action

**Review by seeflood — APPROVED** [review]
lgtm

File: .github/workflows/pr-title-checker.yml [hunk 1]
```diff
@@ -14,8 +14,20 @@ jobs:
     name: "🍀 Title Validation"
     runs-on: ubuntu-latest
     steps:
-      - uses: thehanimo/pr-title-checker@v1.3.4
-        with:
+      - uses: amannn/action-semantic-pull-request@v2.1.0
```
Comment (seeflood) [inline .github/workflows/pr-title-checker.yml:17]: @Xunzhuo How about using the latest version?

**Comment by seeflood** [comment]
@Xunzhuo Hi, as a poor reader, I didn't find out how to grant access to the TOKEN after reading the doc 😭 
Does the action automatically get permission after being merged?

**Review by zhenjunMa — APPROVED** [review]

**Pull request merged**